src/v/kafka/client/test/reconnect.cc

```diff
@@ -74,3 +76,78 @@ FIXTURE_TEST(reconnect, kafka_client_fixture) {
     info("Stopping client");
     client.stop().get();
 }
+inline http::client make_admin_client() {
+    net::base_transport::configuration transport_cfg;
+    transport_cfg.server_addr = net::unresolved_address{"127.0.0.1", 9644};
+    return http::client(transport_cfg);
+}
+
+FIXTURE_TEST(password_change_live_client, kafka_client_fixture) {
+    ss::sstring username{"admin"};
+    ss::sstring userpass{"foopar"};
+
+    info("Waiting for leadership");
+    wait_for_controller_leadership().get();
+
+    info("Enable SASL and restart");
+    enable_sasl_and_restart(username);
+
+    ss::sstring user_body = fmt::format(
+      R"({{"username": "{}", "password": "{}","algorithm": "SCRAM-SHA-256"}})",
+      username,
+      userpass);
+    auto body = iobuf();
+    body.append(user_body.data(), user_body.size());
+
+    info("Create superuser");
+    auto admin_client = make_admin_client();
+    auto res = http_request(
+      admin_client,
+      "/v1/security/users",
+      std::move(body),
+      boost::beast::http::verb::post);
+    BOOST_REQUIRE_EQUAL(res.headers.result(), boost::beast::http::status::ok);
+
+    auto tp = model::topic_partition(model::topic("t"), model::partition_id(0));
+    auto kafka_client = make_client();
+    kafka_client.config().sasl_mechanism.set_value(
+      ss::sstring{"SCRAM-SHA-256"});
+    kafka_client.config().scram_username.set_value(username);
+    kafka_client.config().scram_password.set_value(userpass);
+    kafka_client.connect().get();
+
+    {
+        info("Adding known topic");
+        auto ntp = make_default_ntp(tp.topic, tp.partition);
+        add_topic(model::topic_namespace_view(ntp)).get();
+    }
+
+    {
+        info("Checking for known topic");
+        auto res = kafka_client.dispatch(make_list_topics_req()).get();
+        BOOST_REQUIRE_EQUAL(res.data.topics.size(), 1);
+        BOOST_REQUIRE_EQUAL(res.data.topics[0].name(), "t");
+    }
+
+    {
+        // Setting the password has no effect until the client disconnects
+        info("Changing password");
+        userpass = "foobar";
+        kafka_client.config().scram_password.set_value(userpass);
+    }
+
+    {
+        info("Recheck for known topic");
+        auto res = kafka_client.dispatch(make_list_topics_req()).get();
+        BOOST_REQUIRE_EQUAL(res.data.topics.size(), 1);
+        BOOST_REQUIRE_EQUAL(res.data.topics[0].name(), "t");
+    }
+
+    info("Stopping kafka client");
+    kafka_client.stop().get();
+
```

Comment:
Should the test start up a new kafka client with the old password and ensure it fails? Or a new client with the new password and ensure it works? Or are those covered somewhere else?

Reply:
I think that is/will-be a different test and doesn't need to block this PR. But you make a good point

Reply:
> Or a new client with the new password and ensure it works?

This one is in our unit tests on the Pandaproxy side https://github.com/redpanda-data/redpanda/blob/8bbc13dea30a92edc119e57661021143f137d96f/src/v/pandaproxy/test/kafka_client_cache.cc#L102 so I think we can leave this as is for now.